pkg/operation/botanist/dns.go

```diff
@@ -126,63 +129,55 @@ func (b *Botanist) DefaultExternalDNSProvider(seedClient client.Client) componen
 					Exclude: b.Shoot.ExternalDomain.ExcludeZones,
 				},
 			},
-			b.Shoot.SeedNamespace,
-			b.K8sSeedClient.ChartApplier(),
-			b.ChartsRootPath,
-			b.Logger,
-			seedClient,
 			nil,
 		)
 	}
 
-	return component.OpDestroy(dns.NewDNSProvider(
+	return component.OpDestroy(dns.NewProvider(
+		b.Logger,
+		seedClient,
+		b.Shoot.SeedNamespace,
 		&dns.ProviderValues{
 			Name:    DNSExternalName,
 			Purpose: DNSExternalName,
 		},
-		b.Shoot.SeedNamespace,
-		b.K8sSeedClient.ChartApplier(),
-		b.ChartsRootPath,
-		b.Logger,
-		seedClient,
 		nil,
 	))
 }
 
 // DefaultExternalDNSEntry returns DeployWaiter which removes the external DNSEntry.
 func (b *Botanist) DefaultExternalDNSEntry(seedClient client.Client) component.DeployWaiter {
-	return component.OpDestroy(dns.NewDNSEntry(
+	return component.OpDestroy(dns.NewEntry(
+		b.Logger,
+		seedClient,
+		b.Shoot.SeedNamespace,
 		&dns.EntryValues{
 			Name: DNSExternalName,
 			TTL:  *b.Config.Controllers.Shoot.DNSEntryTTLSeconds,
 		},
-		b.Shoot.SeedNamespace,
-		b.K8sSeedClient.ChartApplier(),
-		b.ChartsRootPath,
-		b.Logger,
-		seedClient,
 		nil,
 	))
 }
 
 // DefaultExternalDNSOwner returns DeployWaiter which removes the external DNSOwner.
 func (b *Botanist) DefaultExternalDNSOwner(seedClient client.Client) component.DeployWaiter {
-	return component.OpDestroy(dns.NewDNSOwner(
+	return component.OpDestroy(dns.NewOwner(
+		seedClient,
+		b.Shoot.SeedNamespace,
 		&dns.OwnerValues{
 			Name: DNSExternalName,
```

Comment:
Active used to be defaulted to `true` now it is defaulted to `false` this could introduce a change here, right?

Reply:
Very nice catch, thanks a lot. I will check and adapt!

Reply:
Done with 115ec34, have a look